app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.kt

```diff
@@ -310,7 +310,7 @@ object Utils {
 
 }
 
-fun isAllGranted(grantResults: Map<String?, Boolean?>) = grantResults.containsValue(false)
+fun isAllGranted(grantResults: Map<String?, Boolean?>) = grantResults.containsValue(true)
```

Comment:
We need to check for every result.
You should use:
```suggestion
fun isAllGranted(grantResults: Map<String?, Boolean?>) = !grantResults.containsValue(false)
```